src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -183,7 +183,7 @@ public static partial class JsonSerializer
                     else if (bytesInBuffer != 0)
                     {
                         // Shift the processed bytes to the beginning of buffer to make more room.
-                        Buffer.BlockCopy(buffer, bytesConsumed, buffer, 0, bytesInBuffer);
+                        Buffer.BlockCopy(buffer, bytesConsumed + start, buffer, 0, bytesInBuffer);
```

Comment:
Is it guaranteed here that `bytesConsumed + start` will be <= buffer.Length and same with `bytesInBuffer + bytesConsumed + start` (and also won't overflow int32)? Add debug.asserts.

Reply:
This issue didn't change any of that logic; also BlockCopy will validate and throw ArgumentException in those cases so I don't see the need to duplicate that check for an Assert here.